web/web/auth/auth.py

```diff
@@ -9,5 +13,42 @@
 
 @auth_bp.route('/')
 @auth_bp.route('/auth', strict_slashes=False)
+@jwt_optional
 def index():
-    return render_template('auth/login.html')
+    access_token = request.cookies.get('access_token')
+    refresh_token = request.cookies.get('refresh_token')
+    if access_token and refresh_token:
+        return redirect('power/capacity')
+    else:
+        return render_template('auth/login.html')
+
+
+# see this repo for JWT redis blacklist setup:
+# https://github.com/vimalloc/flask-jwt-extended/blob/master/examples/redis_blacklist.py
+
+
+# Revokes the current user's access token
+@auth_bp.route('/auth/access_revoke', methods=['DELETE'])
+@jwt_required
+def logout():
+    jti = get_raw_jwt()['jti']
+    revoked_store.set(jti, 'true', timedelta(minutes=15) * 1.2)
+    return jsonify({"msg": "Access token revoked"}), 200
+
+# Revokes the current user's refresh token
+@auth_bp.route('/auth/refresh_revoke', methods=['DELETE'])
+@jwt_refresh_token_required
+def logout2():
+    jti = get_raw_jwt()['jti']
+    revoked_store.set(jti, 'true', timedelta(days=30) * 1.2)
+    return jsonify({"msg": "Refresh token revoked"}), 200
```

Comment:
I know this is in the example link above, but we can dry up this endpoint into one.

Also, pull in the timedelta value from the config.

Reply:
So in order to revoke the access token and refresh token in the same endpoint, since I can't pass both tokens in the Authorization header, is to pass only the access token in the header, and then get the refresh token from the cookies inside the endpoint. I am a little hesitant to implement this work around approach. The Flask-JWT-Ext docs also have the two different logout formats. Let me know what you think and if you have a good resource to share! https://flask-jwt-extended.readthedocs.io/en/stable/blacklist_and_token_revoking/